### DISCUSSION TITLE
CoastalME Tools

### DISCUSSION THREAD

**Discussion by davefavismortlock** [coastalme/coastalme#6] [General]
In London a couple of weeks ago, Wilf and Tom showed off the rather excellent CoastalME Tools. Great! But some questions...

1. What about feature requests for CoastalME Tools? For example, could CoastalME Tools have an option to search the CoastalME log file for any messages relating to the currently-viewed timestep?

2. When can the rest of us get our hands on CoastalME Tools?

3. Should there be some connection made between the CoastalME GitHub site and CoastalME Tools? I think Wilf said that CoastalME Tools is in a private GitHub repository, so I understand that we can't make a link while the Tools repository is private.

Dave

**Comment by wilfchun** [comment]
Hi Dave, this is all currently a work in progress from our end.
In an ideal world I would like to make the repo public, and then operate it with feature requests similar to CoastalMe. This way it can also be linked with the main repo if that is what we would like to do. Unfortunately, we just need to go thorough a few hoops before we do this.

I will keep you and the rest of the team posted with any updates.

**Reply by davefavismortlock** [reply]
Again, fair enough. If there is anything I can do to help with Tools, then just shout.

Dave

**Reply by davefavismortlock** [reply]
And please consider my feature request, above. That would be really really useful :-)

**Reply by wilfchun** [reply]
I have enabled it to scan the log file for any entries containing ERROR or WARNING and (colour) print them after a cme run, examples are shown below:
![image](https://github.com/user-attachments/assets/6fb16a1f-5488-4ecc-8418-09d98387bdaf)
![image](https://github.com/user-attachments/assets/ae823d78-42a7-408a-8fe8-a5767fd5fa40)
